homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,463 @@
+"""Support for Tuya Climate."""
+
+from __future__ import annotations
+import json
+import logging
+
+from typing import Any
+from tuya_iot import TuyaDevice, TuyaDeviceManager
+
+from homeassistant.components.climate import DOMAIN as DEVICE_DOMAIN, ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_FAN,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    CURRENT_HVAC_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .base import TuyaHaEntity
+from .const import (
+    DOMAIN,
+    TUYA_DEVICE_MANAGER,
+    TUYA_DISCOVERY_NEW,
+    TUYA_HA_DEVICES,
+    TUYA_HA_TUYA_MAP,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# Air Conditioner
+# https://developer.tuya.com/en/docs/iot/f?id=K9gf46qujdmwb
+DPCODE_SWITCH = "switch"
+DPCODE_TEMP_SET = "temp_set"
+DPCODE_TEMP_SET_F = "temp_set_f"
+DPCODE_MODE = "mode"
+DPCODE_HUMIDITY_SET = "humidity_set"
+DPCODE_FAN_SPEED_ENUM = "fan_speed_enum"
+
+# Temperature unit
+DPCODE_TEMP_UNIT_CONVERT = "temp_unit_convert"
+DPCODE_C_F = "c_f"
+
+# swing flap switch
+DPCODE_SWITCH_HORIZONTAL = "switch_horizontal"
+DPCODE_SWITCH_VERTICAL = "switch_vertical"
+
+# status
+DPCODE_TEMP_CURRENT = "temp_current"
+DPCODE_TEMP_CURRENT_F = "temp_current_f"
+DPCODE_HUMIDITY_CURRENT = "humidity_current"
+
+SWING_OFF = "swing_off"
+SWING_VERTICAL = "swing_vertical"
+SWING_HORIZONTAL = "swing_horizontal"
+SWING_BOTH = "swing_both"
+
+TUYA_HVAC_TO_HA = {
+    "hot": HVAC_MODE_HEAT,
+    "cold": HVAC_MODE_COOL,
+    "wet": HVAC_MODE_DRY,
+    "wind": HVAC_MODE_FAN_ONLY,
+    "auto": HVAC_MODE_AUTO,
```

Comment:
Is this an auto mode where the user can't change temperature?

Reply:
Yes